diagnostic_aggregator/src/aggregator.cpp

```diff
@@ -205,6 +210,15 @@ void Aggregator::publishData()
     }
   }
 
+  // check if we have changed stamp before we populate the stamp
+  // so that we can simply compare messages
+  if (diag_array != previous_aggregation_) {
+    agg_stateful_pub_->publish(diag_array);
+  } else {
+    RCLCPP_DEBUG(logger_, "No change in aggregation, not publishing");
+  }
+  previous_aggregation_ = diag_array;
```

Comment:
If we want to check equality of content, rather than the stamp, perhaps a more robust way to do this would just be to compare the `status` field, which is overloaded to still do deep equality.

Reply:
good point, agreed. I will take care of it

Reply:
although wait -- I am not checking the stamp itself. Maybe the comment needs to be rewritten to be clearer.

Look again, I am comparing the messages deeply, I am just comparing the version that has not been stamped yet.

Reply:
Yes, I understand you are comparing deeply, but if you compared the status field, you wouldn't have to worry about when you assign the stamp. Just a thought: I think it's fine as it is.